backend/src/migrations/004_comment_response_refactor.ts

```diff
@@ -0,0 +1,73 @@
+import AccessRequestModel from '../models/AccessRequest.js'
+import ReleaseModel from '../models/Release.js'
+import ResponseModel, { ResponseKind } from '../models/Response.js'
+import ReviewModel from '../models/Review.js'
+import { toEntity } from '../utils/entity.js'
+
+export async function up() {
+  // Resolve release comments
+  const releases: any = await ReleaseModel.find({})
+  for (const release of releases) {
+    release.commentIds = []
+    if (release.get('comments') !== undefined) {
+      for (const releaseComment of release.get('comments')) {
+        const newComment = new ResponseModel({
+          user: toEntity('user', releaseComment.user),
+          kind: ResponseKind.Comment,
+          comment: releaseComment.message,
+          createdAt: releaseComment.createdAt,
+        })
+        await newComment.save()
+        release.commentIds.push(newComment._id)
+        release.set('comments', undefined, { strict: false })
+      }
+      await release.save()
+    }
+  }
+
+  // Resolve access request comments
+  const accessRequests: any = await AccessRequestModel.find({})
+  for (const accessRequest of accessRequests) {
+    accessRequest.commentIds = []
+    if (accessRequest.get('comments') !== undefined) {
+      for (const accessCommment of accessRequest.get('comments')) {
+        const newComment = new ResponseModel({
+          user: toEntity('user', accessCommment.user),
+          kind: ResponseKind.Comment,
```

Comment:
It might be useful to have whether it's a comment from an 'access request' or a 'release' on this model.  At the moment it's tricky to go from a comment to the object it's placed on, which might be a useful operation.  I'm less sure on whether this is a good or bad idea, so will raise it with the rest of the team.

Reply:
I'm conflicted - I was trying to keep it simple by not having this two way relationship between the two models, and just having releases/access requests store a list of commentIds. 

Can you think of a useful use case of where we'd want to go to the release or access request from the comment itself? I think from a UI perspective we don't have comments on their own page anywhere, they are only displayed on the release or access request page. 

We also might want to add comments to model cards or other places in the future which would increase complexity if we were to go with your approach, whereas it would work "out of the box" in its current format.

I'm happy to be persuaded to change if the rest of the team think it's useful, though. :)